store/src/java/com/zimbra/cs/service/mail/ModifyCalendarItem.java

```diff
@@ -108,8 +109,8 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         MailSendQueue sendQueue = new MailSendQueue();
         Element response = getResponseElement(zsc);
         int compNum = (int) request.getAttributeLong(MailConstants.A_CAL_COMP, 0);
-        mbox.lock.lock();
-        try {
+        try (final MailboxLock l = mbox.lockFactory.writeLock()) {
```

Comment:
Change this to `mbox.lock(true)` instead of `mbox.lockFactory.writeLock()`

Reply:
Including the mailbox lock in `try-with-resources` here changes the semantics of this operation. Be careful not to do this.

Notice that `sendQueue.send()` will now occur before the `MailboxLock` is closed.

You are going to need to wrap the body of this try statement with another `try-with-resources` that manages the lifecycle of the `MailboxLock`